apps/andi/lib/andi_web/live/ingestion_live_view/edit_ingestion_live_view.ex

```diff
@@ -82,27 +111,38 @@ defmodule AndiWeb.IngestionLiveView.EditIngestionLiveView do
     """
   end
 
-  def mount(
-        _params,
-        %{"is_curator" => is_curator, "ingestion" => ingestion, "user_id" => user_id} = _session,
+  def handle_info(
+        {:updated_metadata,
+          %Ecto.Changeset{ data: %AndiWeb.InputSchemas.IngestionMetadataFormSchema{} } = metadata_changeset },
         socket
       ) do
-    default_changeset = InputConverter.andi_ingestion_to_full_ui_changeset(ingestion)
+    new_ingestion_changeset = Andi.InputSchemas.Ingestion.merge_metadata_changeset(socket.assigns.changeset, metadata_changeset)
```

Comment:
This handler catches child messages with the :updated_metadata event. It then merges the child metadata into the parent changeset which:
- Performs any validation at the top/api level
- Keeps state in sync with other children
- Rerenders the direct child **and any other children that share information!** No need to broadcast messages between children.

Note: We unfortunately still have to broadcast about the source-format because data_dictionary_form.ex is not refactored into this parent/child pattern yet.

Reply:
Many wins in this refactor from Ryan but "no need to broadcast messages between children" is big on the list for me